File: HW/HW7/hw7.java
package HW.HW7;

import java.util.Arrays;

public class hw7 {
    public int minMeetingRooms(int[][] intervals) {  

        int count = 0; 
        int max = 0; 
        int[] start = new int[intervals.length];
        int[] end = new int[intervals.length];

        for (int i = 0; i < intervals.length; i++) { 
            start[i] = intervals[i][0];
            end[i] = intervals[i][1];
        }
        Arrays.sort(start);
        Arrays.sort(end);

        int l = 0;
        int r = 0; 

        while (l < intervals.length) { 
            if (start[l] < end[r]) { 
                count++; 
                l++;
            } else { 
                r++;
                count--;
            }
            max = (max > count) ? max : count;
        }
        
        return max;
     }

     public static void main(String[] args) {
        int[][] test1 = { {0,30},{5,10},{15,20} };
        int[][] test2 = { {0, 1}, {1, 2}, {2, 3}};
        int[][] test3 = { {7, 10}, {2, 4} };

        hw7 x = new hw7();

        System.out.println("Test 1: " + x.minMeetingRooms(test1));
        System.out.println("Test 2: " + x.minMeetingRooms(test2));
        System.out.println("Test 3: " + x.minMeetingRooms(test3));

     }

}

File: HW/HW7/hw7.py
from typing import List

class hw2: 
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:   
        
        start = []
        end = [] 
        
        for i in range(len(intervals)):
            start.append(intervals[i][0])
            end.append(intervals[i][1])

        start.sort()
        end.sort()

        result = 0
        count = 0

        l = 0
        r = 0

        while l < len(intervals):
            if start[l] < end[r]: 
                count += 1
                l += 1
            else:
                count -= 1
                r += 1
            result = max(result, count)

        return result

x = hw2()

test1 = ( [0,30],[5,10],[15,20] )
test2 = ( [0, 1], [1, 2], [2, 3] )
test3 = ( [7, 10], [2, 4] )

print("Test 1: ", x.minMeetingRooms(test1))
print("Test 2: ", x.minMeetingRooms(test2))
print("Test 3: ", x.minMeetingRooms(test3))


